File: 8.字符串转换整数/Change.java
class Solution {
    public int myAtoi(String s) {
        String s1 = s.strip();
        if(s1.equals("")) return 0;
        char sub = '+';
        if(s1.charAt(0)=='-'){
            sub = '-'; 
            s1 = s1.substring(1);
        }else if(s1.charAt(0)=='+'){
            sub = '+'; 
            s1 = s1.substring(1);
        }
        
        if(s1.equals("")||!(s1.charAt(0)>=48&&s1.charAt(0)<=57)){
            return 0;
        }
        String temp="";
        for(char a:s1.toCharArray()){       
            try {
                Integer.parseInt(temp+a);
            }catch(Exception e){
                if(a>=48&&a<=57){
                    if(sub=='-') return -2147483648;
                    return 2147483647;
                }else{
                    if(sub=='-') return -Integer.parseInt(temp);
                    return Integer.parseInt(temp);
                }
            }
            temp = temp + a;
        }
        if(sub=='-') return -Integer.parseInt(temp);
        return Integer.parseInt(temp);
    }
}

File: 8.字符串转换整数/Change.c
class Solution {
    public static int myAtoi(String s) {
        s = s.trim();
        s = s + "a";
        char[] mys = s.toCharArray();
        int i = 0; boolean fu = false;
        if(mys[0]=='-') {i = 1; fu=true;}
        if(mys[0]=='+') {i = 1;}
        while(mys[i]=='0') i++;
        s = s.substring(i);  
        mys = s.toCharArray();
        i = 0;
        while(	(i<=10)&&(i<mys.length)&&
        		(mys[i]=='0'||
        		mys[i]=='1'||
        		mys[i]=='2'||
        		mys[i]=='3'||
        		mys[i]=='4'||
        		mys[i]=='5'||
        		mys[i]=='6'||
        		mys[i]=='7'||
        		mys[i]=='8'||
        		mys[i]=='9')) {
        	i++;
        }
        int end = 0;
        s = s.substring(0,i);
        try {
        	if(i!=0) end = Integer.parseInt(s);
        }catch(Exception e) {
        	if(!fu)	end = 2147483647;
        	else end = -2147483648;
        	fu = false;
        }
        if(fu) end = -end;
        return end;
    }
}